java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -1240,7 +1241,7 @@ protected final void handleSafe(int index, int dataLength) {
      * So even though we may have setup an initial capacity of 1024
      * elements in the vector, it is quite possible
      * that we need to reAlloc() the data buffer when we are setting
-     * the 5th element in the vector simply because previous
+     * the 1025th element in the vector simply because previous
```

Comment:
No, I don't think this change is right. You should read this example as:
* the binary/string vector is 1024 elements long
* the 4 first binary/string elements already occupy 1024 *bytes* in the data buffer, so need to resize the data buffer as soon as the 5th binary/string element is appended

Reply:
Thanks for the detailed explanation. Reverted the change.